test/tall/other/format_width.unit

```diff
@@ -0,0 +1,118 @@
+40 columns                              |
+### Tests for the comment to set formatting width.
+>>> Comment sets page width.
+main() {
+  // dart format width=30
+  fitsUnsplitAt40   +   butNotAt30;
+}
+<<<
+main() {
+  // dart format width=30
+  fitsUnsplitAt40 +
+      butNotAt30;
+}
+>>> Comment anywhere affects all code.
```

Comment:
I don't think this is a useful behavior. Would it be difficult to restrict this to comments before directives?

Reply:
I thought it would, but it turned out to be easy.

I was thinking it might be hard because the formatter supports formatting individual statements too, but there was an easy place to slot it in.

The rule is that the comment must appear before the first non-comment token (basically the first token in the AST being formatted).